iModelCore/ECDb/Tests/NonPublished/ECSqlStatementTests.cpp

```diff
@@ -11800,4 +11800,460 @@ TEST_F(ECSqlStatementTestFixture, SelectAnySomeAll)
+            </ECStructClass>
+        </ECSchema>
+    )xml")));
+
+    ///*** Insertable data
+    auto i = 123;
+    auto p2d = DPoint2d::From(23.22, 31.11);
+
+    auto st_p2d = DPoint2d::From(53.22, 31.11);
+
+    auto st_st_b = true;
+    auto st_st_i = 45;
+
+    //*** Update all values to null
+    {
+    Utf8String ecsql;
+    ecsql.Sprintf("INSERT INTO ts.TestClass (p2d, st, array_i) VALUES (?,?,?)");
+    ECSqlStatement stmt;
+    ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, ecsql.c_str())) << ecsql.c_str();
+    stmt.BindPoint2d(1, p2d);
+    auto& st = stmt.GetBinder(2);
+
+    st["p2d"].BindPoint2d(st_p2d);
+    st["st"]["b"].BindBoolean(st_st_b);
+    st["st"]["i"].BindBoolean(st_st_i);
+
+    stmt.GetBinder(3).AddArrayElement().BindInt(i);
+
+    ECInstanceKey key;
+    ASSERT_EQ(BE_SQLITE_DONE, stmt.Step(key)) << stmt.GetECSql();
+    stmt.Finalize();
+
+    ecsql.Sprintf("UPDATE ONLY ts.TestClass SET p2d=?, st=?, array_i=? WHERE ECInstanceId=%s", key.GetInstanceId().ToString().c_str());
+    ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, ecsql.c_str())) << ecsql.c_str();
+    for (int i = 1; i <= 3; ++i)
+        {
+        ASSERT_EQ(ECSqlStatus::Success, stmt.GetBinder(i).BindNull());
+        }
+
+    ASSERT_EQ(BE_SQLITE_DONE, stmt.Step()) << stmt.GetECSql();
+    stmt.Finalize();
+
+    ecsql.Sprintf("SELECT p2d, p2d.X, p2d.Y, st, array_i FROM ts.TestClass WHERE ECInstanceId=%s", key.GetInstanceId().ToString().c_str());
+    ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, ecsql.c_str())) << ecsql.c_str();
+    ASSERT_EQ(BE_SQLITE_ROW, stmt.Step()) << stmt.GetECSql();
+    for (int i = 0; i < stmt.GetColumnCount(); ++i)
+        {
+        ASSERT_TRUE(stmt.IsValueNull(i)) << "no values bound to " << stmt.GetECSql();
+        }
+
+    const int expectedMembersCount = (int) m_ecdb.Schemas().GetClass("TestSchema", "ComplexStruct")->GetPropertyCount(true);
+    IECSqlValue const& structVal = stmt.GetValue(3);
+    int actualMembersCount = 0;
+    for (IECSqlValue const& memberVal : structVal.GetStructIterable())
+        {
+        actualMembersCount++;
+        ASSERT_TRUE(memberVal.IsNull());
+        }
+    ASSERT_EQ(expectedMembersCount, actualMembersCount);
+
+    IECSqlValue const& structArrayVal = stmt.GetValue(4);
+    ASSERT_EQ(0, structArrayVal.GetArrayLength());
+    }
+
+    //*** Update array to contain two null elements
+    {
+    Utf8String ecsql;
+    ecsql.Sprintf("INSERT INTO ts.TestClass (array_i) VALUES (?)");
+    ECSqlStatement stmt;
+    ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, ecsql.c_str())) << ecsql.c_str();
+    stmt.GetBinder(1).AddArrayElement().BindInt(i);
+
+    ECInstanceKey key;
+    ASSERT_EQ(BE_SQLITE_DONE, stmt.Step(key)) << stmt.GetECSql();
+    stmt.Finalize();
+
+    ecsql.Sprintf("UPDATE ONLY ts.TestClass SET array_i=? WHERE ECInstanceId=%s", key.GetInstanceId().ToString().c_str());

```

Comment:
For anything that goes into SQL we use SqlPrintfString instead

Reply:
Oh, and for specifying an ECInstanceId like you do here and several times more in the method, it's better to not printf it into the statement but use stmt.BindId() instead

Reply:
As discussed stip the printf calls which do not have any parameter, then this conversation can be marked as resolved. Thanks

Reply:
Sorted that out as well.